### PR TITLE
Do not fail health check when schema is in agreement with storage nodes

### PR DESCRIPTION
**Pull request by dimas-b**

### CODE REVIEW

File: persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java [hunk 1]
```diff
@@ -309,6 +310,25 @@ public boolean isInSchemaAgreement() {
         <= 1;
   }
 
+  @Override
+  public boolean isInSchemaAgreementWithStorage() {
```
Comment (ivansenic) [inline persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:314]: No extra tests for these new `isInSchemaAgreementWithStorage` implementations?

**Review by ivansenic — APPROVED** [review]
Seems like exactly what we need..

File: persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java [hunk 2]
```diff
@@ -309,6 +310,25 @@ public boolean isInSchemaAgreement() {
...
+    InetAddress localAddress = FBUtilities.getBroadcastAddress();
+    return Gossiper.instance.getLiveMembers().stream()
+            .filter(
+                ep -> {
+                  EndpointState epState = Gossiper.instance.getEndpointStateForEndpoint(ep);
+                  return epState != null
+                      && !Gossiper.instance.isDeadState(epState)
+                      && (!Gossiper.instance.isGossipOnlyMember(ep) || localAddress.equals(ep));
+                })
+            .map(Gossiper.instance::getSchemaVersion)
```
Comment (tomekl007) [inline persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:326]: Why we need this map stage? As far as I can see, the returned `schemaVersion` value is not checked anywhere.
There is only a check on `size <= 1;`.

File: persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java (hunk 1 above)
Comment (tomekl007) [reply persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:314]: +1 on adding tests

Comment (tomekl007) [reply persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:314]: +1 on adding test(s)

File: persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java (hunk 2 above)
Comment (dimas-b) [reply persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:326]: We collect schema UUIDs in a set in the following step to remove duplicates. The size check ensures at most one UUID remains.

Comment (dimas-b) [reply persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:326]: We could refactor this to use some other technique to achieve the same result. I just followed the pattern of other similar methods that check schema versions.

File: persistence-api/src/main/java/io/stargate/db/Persistence.java [hunk 3]
```diff
@@ -86,6 +86,11 @@
 
   boolean isInSchemaAgreement();
 
+  /** Returns <code>true</code>> if the local schema agrees with all storage nodes. */
```
[the remark this answers is not in the document]
Comment (dimas-b) [reply persistence-api/src/main/java/io/stargate/db/Persistence.java:89]: You're right. I'll remove the default impl.

File: persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java (hunk 2 above)
Comment (tomekl007) [reply persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:326]: >We collect schema UUIDs in a set in the following step to remove duplicates. The size check ensures at most one UUID remains.
It seems implicit, I am +1 on refactoring that if that's not a big effort.

Comment (dimas-b) [reply persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:326]: refactored

File: persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java (hunk 1 above)
Comment (dimas-b) [reply persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java:320]: tests added

**Review by dougwettlaufer — APPROVED** [review]

**Pull request merged**